packages/cli/src/deploy/getFunctions.ts

```diff
@@ -0,0 +1,58 @@
+import { Client, getFunctionSelector, parseAbiItem } from "viem";
+import { WorldDeploy, WorldFunction, worldTables } from "./common";
+import { debug } from "./debug";
+import { storeSetRecordEvent } from "@latticexyz/store";
+import { getLogs } from "viem/actions";
+import { decodeValueArgs } from "@latticexyz/protocol-parser";
+import { getTableValue } from "./getTableValue";
+import { hexToResource } from "@latticexyz/common";
+
+export async function getFunctions({
+  client,
+  worldDeploy,
+}: {
+  readonly client: Client;
+  readonly worldDeploy: WorldDeploy;
+}): Promise<readonly WorldFunction[]> {
+  // This assumes we only use `FunctionSelectors._set(...)`, which is true as of this writing.
```

Comment:
Since we don't have any sort of "index" for tables like FunctionSelectors, we look at all the keys used in the logs for the table. But since viem can only filter on one event+args (could be an RPC node limitation), we can't easily fetch all store logs for a given table ID and recompute the current state of the table. So we just use the one event that is actually used internally.

Reply:
got it, thanks for the context